### PR TITLE
[O2-5295] DPL Analysis: introduce aod-writer-df-offset option

### PR DESCRIPTION
**Pull request by aalkin**
Introduces `--aod-writer-df-offset N` option that makes dataframes numbers start from `N+1`. Can also be set in writer configuration JSON file as 'offset'.

### CODE REVIEW

File: Framework/Core/src/DataOutputDirector.cxx [hunk 1]
```diff
@@ -351,8 +350,21 @@ std::tuple<std::string, std::string, std::string, float, int> DataOutputDirector
...
+      setDFOffset(offset);
+    } else {
+      LOGP(error, "Check the JSON document! Item \"{}\" must be a number!", itemName);
+      return memptyanswer;
+    }
+  }
+
   itemName = "OutputDescriptors";
   if (dodirItem.HasMember(itemName)) {
+    std::string slh("/");
```
Comment (ktf) [inline Framework/Core/src/DataOutputDirector.cxx:366]: static / const, no?

**Review by ktf — DISMISSED** [review]
Assuming it was tested, fine with me. We should get rid of the tuple in a subsequent PR, IMHO.

**Review by sawenzel — DISMISSED** [review]

File: Framework/Core/src/DataOutputDirector.cxx (hunk 1 above)
Comment (aalkin) [reply Framework/Core/src/DataOutputDirector.cxx:366]: The whole thing needs a refactoring pass, but it is not critical.

**Comment by jgrosseo** [comment]
I just posted comments on the related JIRA ticket which may be relevant...

The DF/TF number is internally available and some task may grab it and use it for something. In this case the number will be different in the file than during the current processing. Why we do not set it before generation?

**Comment by jgrosseo** [comment]
Discussion continues on https://its.cern.ch/jira/browse/O2-5295

**Comment by aalkin** [comment]
Replaced by #13711

**Pull request closed without merging**